components/autocomplete/index.js

```diff
@@ -221,9 +221,15 @@ export class Autocomplete extends Component {
 		}
 	}
 
-	loadOptions( index ) {
-		this.props.completers[ index ].getOptions().then( ( options ) => {
-			const keyedOptions = map( options, ( option, i ) => ( { ...option, key: index + '-' + i } ) );
+	/**
+	 * Load options for an autocompleter.
+	 *
+	 * @param  {int} index    The autocompleter index.
+	 * @param  {string} query The query, if any.
+	 */
+	loadOptions( index, query ) {
+		this.props.completers[ index ].getOptions( query ).then( ( options ) => {
+			const keyedOptions = map( options, ( option, i ) => ( { ...option, key: ( option.value.slug ? option.value.slug : index ) + '-' + i } ) );
```

Comment:
I think you  might have the `index` and `i` the wrong way around, I think it should be:
```javascript
key: index + '-' + ( option.value.slug ? option.value.slug : i )
```
This is because `index` should be constant for the completer.

Reply:
I fixed this in https://github.com/WordPress/gutenberg/pull/4406/commits/eee4ce31ee32b396ef4a788e60ac0c0cca515a9b